lib/grp.gi

```diff
@@ -241,13 +243,15 @@ InstallMethod( IsPGroup,
     # known *and* the group knows to be nilpotent or is abelian;
     # thus an `IsAbelian' test may be forced (which can be done via comparing
     # products of generators) but *not* an `IsNilpotent' test.
-    if     ( not HasSize( G ) )
-       and (    ( HasIsNilpotentGroup( G ) and IsNilpotentGroup( G ) )
-             or IsAbelian( G ) ) then
+    if HasSize( G ) and IsFinite( G ) then
+      return IS_PGROUP_FROM_SIZE( G );
+    elif ( HasIsNilpotentGroup( G ) and IsNilpotentGroup( G ) )
```

Comment:
Something just occurred to me. Does infinite nilpotency imply that a group is a p-group if and only if all generators have p-power orders? If not, then an ````IsFinite```` check is needed here when ````IsPgroup```` will not imply ````IsFinite````. I have not thought about this before. 

Fortunately it does, see Plop's answer in https://math.stackexchange.com/questions/36056/nilpotent-group-is-almost-abelian-counterexample-for-infinite-order-case
According to this, for every nilpotent group two elements with relative prime order must commute. 

Could you maybe add a comment to reflect this, so that people some years from now reading the code would not need to figure this out again?

Reply:
I'll add a comment.